projects/js-packages/jetpack-cli/bin/jp.js

```diff
@@ -0,0 +1,286 @@
+				fs.mkdirSync( resolve( monorepoRoot, 'tools/docker/data/jetpack_dev_mysql' ), {
+					recursive: true,
+				} );
+				fs.mkdirSync( resolve( monorepoRoot, 'tools/docker/data/ssh.keys' ), { recursive: true } );
+				fs.mkdirSync( resolve( monorepoRoot, 'tools/docker/wordpress' ), { recursive: true } );
+
+				const images = [
+					{ name: 'mariadb:lts' },
+					{ name: 'automattic/jetpack-wordpress-dev:latest' },
+					{ name: 'phpmyadmin/phpmyadmin:latest', platform: 'linux/amd64' },
+					{ name: 'maildev/maildev', platform: 'linux/amd64' },
+					{ name: 'atmoz/sftp', platform: 'linux/amd64' },
+				];
+
+				for ( const image of images ) {
+					const inspect = spawnSync( 'docker', [ 'image', 'inspect', image.name ], {
+						stdio: 'ignore',
+					} );
+					if ( inspect.status !== 0 ) {
+						console.log( chalk.blue( `Pulling ${ image.name }...` ) );
+						const pullArgs = [ 'pull', image.name ];
+						if ( image.platform ) {
+							pullArgs.splice( 1, 0, '--platform', image.platform );
+						}
+						const pull = spawnSync( 'docker', pullArgs, { stdio: 'inherit' } );
+						if ( pull.status !== 0 ) {
+							throw new Error( `Failed to pull ${ image.name }` );
+						}
+					}
+				}
+
+				const configResult = spawnSync(
+					resolve( monorepoRoot, 'tools/docker/bin/monorepo' ),
+					[ 'pnpm', 'jetpack', 'docker', 'config' ],
+					{
+						stdio: 'inherit',
+						shell: true,
+						cwd: monorepoRoot,
+					}
+				);
+
+				if ( configResult.status !== 0 ) {
+					throw new Error( 'Failed to generate Docker config' );
+				}
+			}
+
+			// Get project name (from docker.js)
+			const projectName = args.includes( '--type=e2e' ) ? 'jetpack_e2e' : 'jetpack_dev';
+
+			// Load versions from .github/versions.sh
+			const versionsPath = resolve( monorepoRoot, '.github/versions.sh' );
+			const versions = fs.readFileSync( versionsPath, 'utf8' );
+			const versionVars = {};
+			versions.split( '\n' ).forEach( line => {
+				const match = line.match( /^([A-Z_]+)=(.+)$/ );
+				if ( match ) {
+					versionVars[ match[ 1 ] ] = match[ 2 ].replace( /['"]/g, '' );
+				}
+			} );
+
+			// Build environment variables (from docker.js)
+			const envVars = {
+				...process.env,
+				// Load from default.env
+				...( fs.existsSync( resolve( monorepoRoot, 'tools/docker/default.env' ) )
+					? dotenv.parse( fs.readFileSync( resolve( monorepoRoot, 'tools/docker/default.env' ) ) )
+					: {} ),
+				// Load from .env if it exists
+				...( fs.existsSync( resolve( monorepoRoot, 'tools/docker/.env' ) )
+					? dotenv.parse( fs.readFileSync( resolve( monorepoRoot, 'tools/docker/.env' ) ) )
+					: {} ),
+				HOST_CWD: monorepoRoot,
+				PHP_VERSION: versionVars.PHP_VERSION,
+				COMPOSER_VERSION: versionVars.COMPOSER_VERSION,
+				NODE_VERSION: versionVars.NODE_VERSION,
+				PNPM_VERSION: versionVars.PNPM_VERSION,
+				COMPOSE_PROJECT_NAME: projectName,
+				PORT_WORDPRESS: args.includes( '--type=e2e' ) ? '8889' : '80',
```

Comment:
This will work for most people, so maybe it's fine, but I do note it will blow up if something else is already running on localhost port 80.

Reply:
nod, I believe this is the same as the existing `jetpack docker`, except the existing command accepts an optional port arg

Reply:
More importantly, the existing command allows overriding the default via `tools/docker/.env`, while your code here overrides `tools/docker/.env`.

Reply:
Ahh, you're totally right. The `.env` should now work after [5e8d778](https://github.com/Automattic/jetpack/pull/40934/commits/5e8d7789fcb2cc7409b00d23c1682d9872a5a446)